ui/sort.go

```diff
@@ -3,10 +3,47 @@ package ui
 import (
 	"cmp"
 	"slices"
+	"time"
 )
 
-func sortMarkdowns(mds []*markdown) {
-	slices.SortStableFunc(mds, func(a, b *markdown) int {
-		return cmp.Compare(a.Note, b.Note)
-	})
+type sortOrder bool
+
+const (
+	ascending  sortOrder = true
+	descending sortOrder = false
+)
+
+func sortMarkdowns(mds []*markdown, sortByDate bool, ascending sortOrder) {
+	if sortByDate {
+		if ascending {
+			slices.SortStableFunc(mds, func(a, b *markdown) int {
+				return compareTime(a.Modtime, b.Modtime)
+			})
+		} else {
+			slices.SortStableFunc(mds, func(a, b *markdown) int {
+				return -compareTime(a.Modtime, b.Modtime)
+			})
+		}
+	} else {
+		if ascending {
+			slices.SortStableFunc(mds, func(a, b *markdown) int {
+				return cmp.Compare(a.Note, b.Note)
+			})
+		} else {
+			slices.SortStableFunc(mds, func(a, b *markdown) int {
+				return -cmp.Compare(a.Note, b.Note)
+			})
+		}
+	}
+}
```

Comment:
Some suggestions:

1. Instead of repeating `slices.SortStableFunc` 4 times, assigning functions and calling  in the end (make it a one-liner).
2. Change `if/else` into a `switch` case to reduce block inside block.

```suggestion
func sortMarkdowns(mds []*markdown, sortByDate bool, order sortOrder) {
	var fn func(a, b *markdown) int

	switch {
	case sortByDate && order == ascending:
		fn = func(a, b *markdown) int { return compareTime(a.Modtime, b.Modtime) }
	case sortByDate && order == descending:
		fn = func(a, b *markdown) int { return -compareTime(a.Modtime, b.Modtime) }
	case !sortByDate && order == ascending:
		fn = func(a, b *markdown) int { return cmp.Compare(a.Note, b.Note) }
	case !sortByDate && order == descending:
		fn = func(a, b *markdown) int { return -cmp.Compare(a.Note, b.Note) }
	}

	slices.SortStableFunc(mds, fn)
}
```